core/src/main/java/org/testcontainers/containers/DockerComposeContainer.java

```diff
@@ -662,3 +672,54 @@ private Logger logger() {
         return DockerLoggerFactory.getLogger(COMPOSE_EXECUTABLE);
     }
 }
+
+@UtilityClass
+class DockerComposePrecondition {
+    private static final String CONTAINER_NAME_OPTION = "container_name";
+    private static final Yaml YAML = buildYamlParser();
+
+    static void checkNotContainUnsupportedComposeOptions(List<File> composeFiles) {
+        composeFiles.forEach(file -> checkArgument(!containUnsupportedComposeOptions(file),
+            String.format("Compose file %s contains '%s' option which is not supported by container.",
+                file,
+                CONTAINER_NAME_OPTION)));
+    }
+
+    private static boolean containUnsupportedComposeOptions(File file) {
```

Comment:
this does not check for the old format, without the `services` collection

Reply:
As far as i understand docker compose v1 does not provide ability to specify ```container_name``` property at all, which means that it does not make sense to check this for case where ```service``` tag is not specified.

Reply:
https://github.com/docker/compose/blob/7ae632a9ee7530fcf81e212baa3e588f477ea862/compose/config/config_schema_v1.json#L31

that's not true

Reply:
my bad